server/bundles/io.cloudbeaver.server/src/io/cloudbeaver/server/events/WSProjectUpdatedEventHandler.java

```diff
@@ -1,53 +1,49 @@
-/*
- * DBeaver - Universal Database Manager
- * Copyright (C) 2010-2023 DBeaver Corp and others
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
 package io.cloudbeaver.server.events;
 
+import io.cloudbeaver.DBWConstants;
 import io.cloudbeaver.model.session.BaseWebSession;
-import io.cloudbeaver.server.CBPlatform;
 import org.jkiss.code.NotNull;
+import org.jkiss.dbeaver.DBException;
 import org.jkiss.dbeaver.Log;
-import org.jkiss.dbeaver.model.websocket.WSEventHandler;
-import org.jkiss.dbeaver.model.websocket.WSUtils;
-import org.jkiss.dbeaver.model.websocket.event.WSEvent;
-import org.jkiss.utils.CommonUtils;
+import org.jkiss.dbeaver.model.websocket.event.WSEventTopic;
+import org.jkiss.dbeaver.model.websocket.event.WSEventType;
+import org.jkiss.dbeaver.model.websocket.event.WSProjectUpdateEvent;
 
-import java.util.Collection;
+public class WSProjectUpdatedEventHandler extends WSAbstractProjectEventHandler<WSProjectUpdateEvent> {
 
-/**
- * Notify all active user session that project has been updated
- */
-public abstract class WSProjectUpdatedEventHandler implements WSEventHandler {
     private static final Log log = Log.getLog(WSProjectUpdatedEventHandler.class);
 
     @NotNull
     @Override
-    public abstract String getSupportedTopicId();
+    public String getSupportedTopicId() {
+        return WSEventTopic.PROJECTS.getTopicId();
+    }
 
+    @NotNull
     @Override
-    public void handleEvent(@NotNull WSEvent event) {
-        log.debug(getSupportedTopicId() + " event handled");
-        Collection<BaseWebSession> allSessions = CBPlatform.getInstance().getSessionManager().getAllActiveSessions();
-        for (var activeUserSession : allSessions) {
-            if (WSWebUtils.isSessionIdEquals(activeUserSession, event.getSessionId())) {
-                continue;
+    protected Class<WSProjectUpdateEvent> getEventClass() {
+        return WSProjectUpdateEvent.class;
+    }
+
+    @Override
+    protected void updateSessionData(BaseWebSession activeUserSession, WSProjectUpdateEvent event) {
+        if (!activeUserSession.getUserContext().hasPermission(DBWConstants.PERMISSION_ADMIN)) {
```

Comment:
check admin permissions only for RM_PROJECT_CREATED event, when a project is deleted - it must be deleted for all users

Reply:
Fixed